app/grandchallenge/reader_studies/models.py

```diff
@@ -1544,3 +1545,13 @@ def supported_answer_types(self):
                 AnswerType.MULTIPLE_ANGLES,
                 AnswerType.MULTIPLE_ELLIPSES,
             ]
+
+    @classmethod
+    def get_widget_choices_for_answer_type(cls, answer_type):
+        widget_types = []
+        instance = cls()
+        for kind in QuestionWidgetKindChoices:
+            instance.kind = kind
+            if answer_type in instance.supported_answer_types():
+                widget_types.append((kind.name, kind.label))
+        return widget_types
```

Comment:
Maybe there is a better way to do this, but I thought this is at least better than adding a `ANSWER_TYPE_TO_QUESTION_WIDGET_MAP` dictionary which would require maintaining the mapping in two places (the dictionary and in `supported_answer_types`...

Reply:
How about:

```python
ANSWER_TYPE_TO_QUESTION_WIDGET = {
  AnswerType.MULTIPLE_2D_BOUNDING_BOXES: [QuestionWidgetKindChoices.ACCEPT_REJECT, ...]
  ...
}

...

    def supported_answer_types(self):
        return [k for k, v in ANSWER_TYPE_TO_QUESTION_WIDGET.items() if self.kind in v]
```